compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -379,11 +379,11 @@ fn define_all_allocs(tcx: TyCtxt<'_>, module: &mut dyn Module, cx: &mut Constant
             let addend = {
                 let endianness = tcx.data_layout.endian;
                 let offset = offset.bytes() as usize;
-                let ptr_size = tcx.data_layout.pointer_size;
+                let ptr_size = tcx.data_layout.pointer_size.bytes() as usize;
                 let bytes = &alloc.inspect_with_uninit_and_ptr_outside_interpreter(
-                    offset..offset + ptr_size.bytes() as usize,
+                    offset..offset + ptr_size
                 );
-                read_target_uint(endianness, bytes).unwrap()
+                read_target_ptr(endianness, ptr_size, bytes)
```

Comment:
This name makes it seem like it reads the whole pointer instead of just the addend, leaving the target to which the addend is relative out.

Reply:
`read_target_usize` perhaps?

Reply:
Maybe